hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/specifications/TargetSpecifications.java

```diff
@@ -148,6 +150,35 @@ private TargetSpecifications() {
         };
     }
 
+    /**
+     * {@link Specification} for retrieving {@link Target}s by "like attribute
+     * value".
+     *
+     * @param searchText
+     *            to be filtered on
+     * @return the {@link Target} {@link Specification}
+     */
+    public static Specification<JpaTarget> likeAttributeValue(final String searchText) {
+        return (targetRoot, query, cb) -> {
+            final String searchTextToLower = searchText.toLowerCase();
+            final MapJoin<JpaTarget, String, String> attributeMap = targetRoot.join(JpaTarget_.controllerAttributes,
+                    JoinType.LEFT);
+            return cb.or(cb.like(cb.lower(attributeMap.value()), searchTextToLower));
```

Comment:
In order to eliminate duplicates in the search result, you need to set the distinct flag to true:
`query.distinct(true);`

You can easily recreate this by creating a target with a controller ID that is equal to one of the attribute values. In that case the target would appear twice in the list.

Reply:
I've updated the test to check for that, but can't seem to reproduce the issue with it.

Reply:
Thanks for updating the test! I could easily recreate this in the UI a few weeks ago but I will double-check...

Reply:
Hi Stefan,
The new test fails for me! However, if I add query.distinct(true) as proposed above, everything works fine :-) Would be great if you could add that.
Thanks and regards.

Reply:
It seems to pass on CircleCI? Happy to add it, just wondering what's going on there.

Reply:
That's indeed weird. I ran the tests in my Eclipse and there it failed. After adding the distinct flag that tests passed... Anyway, thanks for adding this line!